prisma-fmt/src/references.rs

```diff
@@ -0,0 +1,277 @@
+                };
+
+                find_where_used_in_model(&ctx, arg_value.as_str(), ref_model_id.id.1, ref_model_id.id.0)
+            }
+            _ => empty_identifiers(),
+        },
+
+        // ? This might make more sense to add as a definition rather than a reference
+        SchemaPosition::Model(_, ModelPosition::Field(_, FieldPosition::Attribute(name, _, _)))
+        | SchemaPosition::CompositeType(_, CompositeTypePosition::Field(_, FieldPosition::Attribute(name, _, _))) => {
+            match ctx.datasource().map(|ds| &ds.name) {
+                Some(ds_name) if name.contains(ds_name) => find_where_used_as_top_name(&ctx, ds_name),
+                _ => empty_identifiers(),
+            }
+        }
+
+        SchemaPosition::Model(
+            model_id,
+            ModelPosition::ModelAttribute(_attr_name, _, AttributePosition::ArgumentValue(_, arg_val)),
+        ) => find_where_used_in_model(&ctx, arg_val.as_str(), model_id, ctx.initiating_file_id),
+
+        _ => empty_identifiers(),
+    };
+
+    identifiers
+        .iter()
+        .filter_map(|ident| ident_to_location(ident, &ctx))
+        .collect()
+}
+
+fn find_where_used_in_model<'ast>(
+    ctx: &'ast ReferencesContext<'_>,
+    name: &str,
+    model_id: ModelId,
+    file_id: FileId,
+) -> Vec<&'ast Identifier> {
+    let Some(model) = ctx
+        .db
+        .walk_models()
+        .chain(ctx.db.walk_views())
+        .find(|model| model.id.1 == model_id && model.file_id() == file_id)
+    else {
+        warn!("Could not find model");
+        return empty_identifiers();
+    };
+
+    let identifier = if let Some(field) = model.scalar_fields().find(|field| field.name() == name) {
+        field.ast_field().identifier()
+    } else {
+        warn!("Could not find field with name: `{}`", name);
+        return empty_identifiers();
+    };
+
+    vec![identifier]
+}
+
+fn find_where_used_for_native_type<'ast>(ctx: &ReferencesContext<'ast>, name: &str) -> Vec<&'ast Identifier> {
+    info!("Get references for native types");
+
+    fn find_native_type_locations<'ast>(
+        name: &str,
+        fields: impl Iterator<Item = (FieldId, &'ast Field)>,
+    ) -> Vec<&'ast Identifier> {
+        fields
+            .filter_map(|field| {
+                field
+                    .1
+                    .attributes
+                    .iter()
+                    .find(|attr| extract_ds_from_native_type(attr.name()) == name)
+                    .map(|attr| attr.identifier())
+            })
+            .collect()
+    }
+
+    ctx.db
+        .walk_tops()
+        .flat_map(|top| match top.ast_top() {
+            Top::CompositeType(composite_type) => find_native_type_locations(name, composite_type.iter_fields()),
+            Top::Model(model) => find_native_type_locations(name, model.iter_fields()),
+
+            Top::Enum(_) | Top::Source(_) | Top::Generator(_) => empty_identifiers(),
+        })
+        .collect()
```

Comment:
This could use `ctx.db.find_top` instead which avoids iterating through all tops